src/refiners/foundationals/latent_diffusion/schedulers/ddim.py

```diff
@@ -43,15 +43,14 @@ def __call__(self, x: Tensor, noise: Tensor, step: int) -> Tensor:
                 else tensor(data=[0], device=self.device, dtype=self.dtype)
             ),
         )
-        current_scale_factor, previous_scale_factor = (
-            self.cumulative_scale_factors[timestep],
-            (
-                self.cumulative_scale_factors[previous_timestep]
-                if previous_timestep > 0
-                else self.cumulative_scale_factors[0]
-            ),
+        current_scale_factor, previous_scale_factor = self.cumulative_scale_factors[timestep], (
+            self.cumulative_scale_factors[previous_timestep]
+            if previous_timestep > 0
+            else self.cumulative_scale_factors[0]
         )
         predicted_x = (x - sqrt(1 - current_scale_factor**2) * noise) / current_scale_factor
         denoised_x = previous_scale_factor * predicted_x + sqrt(1 - previous_scale_factor**2) * noise
 
+        self.previous_scale_factor = previous_scale_factor
+
```

Comment:
Why is this re-introduced? It has been removed part of https://github.com/finegrain-ai/refiners/commit/11b0ff6f8c9b79f804b9109bb2c5c2deddc3d11f